cli/compose/loader/loader.go

```diff
@@ -53,11 +53,11 @@ func ParseYAML(source []byte) (map[string]any, error) {
 	if err := yaml.Unmarshal(source, &cfg); err != nil {
 		return nil, err
 	}
-	cfgMap, ok := cfg.(map[any]any)
+	_, ok := cfg.(map[string]any)
```

Comment:
> it seems strange that many tests in the previous PR were failing with top-level object must be a mapping because the tests haven't changed and the compose file used at the time seem ok to me. Even the versions of the module are the same 

So this was  the  reason; YAML v2 returned a  `map[any]any` whereas v3 always returns a `map[string]any`.

Honestly, I half-expected `map[string]any` to _ALSO_ be a `map[any]any`  (because `string` is also an `any` ? But that's not how  Go asserts these.

Reply:
So actually,  to be more correct; it returns a `map[string]interface{}` but Go is happy considering `interface{} == any`